trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/index.ts

```diff
@@ -18,12 +19,18 @@ const renderUtilisationReportReconciliationForReport = (res: Response, viewModel
 export const getUtilisationReportReconciliationByReportId = async (req: Request, res: Response) => {
   const { userToken, user } = asUserSession(req.session);
   const { reportId } = req.params;
-  const { facilityIdQuery } = req.query;
+  const { facilityIdQuery, selectedFeeRecordIds } = req.query;
 
   try {
     const facilityIdQueryAsString = facilityIdQuery ? asString(facilityIdQuery, 'facilityIdQuery') : undefined;
     const facilityIdQueryError = validateFacilityIdQuery(facilityIdQueryAsString, req.originalUrl);
-    const { errorSummary: premiumPaymentFormError, isCheckboxChecked } = getAndClearFieldsFromRedirectSessionData(req);
+
+    const selectedFeeRecordIdsAsString = selectedFeeRecordIds ? asString(selectedFeeRecordIds, 'selectedFeeRecordIds') : undefined;
+    const isCheckboxCheckedQuery = getIsCheckboxCheckedFromQuery(selectedFeeRecordIdsAsString);
+
+    const { errorSummary: premiumPaymentFormError, isCheckboxChecked: isCheckboxCheckedSession } = getAndClearFieldsFromRedirectSessionData(req);
+
+    const isCheckboxChecked = isCheckboxCheckedQuery || isCheckboxCheckedSession;
```

Comment:
This uses the `isCheckboxChecked` arrow func which uses the query params (if defined) and falls back to the session `isCheckboxChecked` func, which itself has a fallback if there's nothing in the session (`() => false`).

I was wondering about moving the default fallback here? eg 
```
const isCheckboxCheckedDefault = () => false;
const isCheckboxChecked = isCheckboxCheckedQuery || isCheckboxCheckedSession || isCheckboxCheckedDefault;
```

Any thoughts on the way its implemented in this PR, or the possible implementation above in this comment.

Reply:
What if we moved each to return this page just a list of checked checkbox ids and then we had a single helper function here that constructs the method from an array?